TUTORIAL.md

```diff
@@ -0,0 +1,392 @@
+
+***Note: `create_table` and `drop_table` are tested together in the specs, so you will have to build both of them as well as add your `attr_accessors` before your tests pass.***
+
+##lib/course.rb
+
+```ruby
+class Course
+attr_accessor :id, :name, :department_id
+
+  def self.create_table
+    sql = <<-SQL
+    CREATE TABLE IF NOT EXISTS courses (
+      id INTEGER PRIMARY KEY,
+      name TEXT,
+      department_id INTEGER 
+    )
+    SQL
+     
+    DB[:conn].execute(sql)
+  end
+  
+end
+
+```
+We now have a way to create a table, but what if we want to drop a table? Yep, we need a `drop_table` method to complete the circuit. Take a look at your `lib/student.rb` and you will see a `drop_table` method for students, use that as your guide. This method is pretty clear, if a table exists, drop it when the method is executed. Just like our `create_table` method, we need to prepend `self`, which accesses our Class scope. Since the courses table is responsible for storing all the courses in the Course class, it makes sense for this to be a class method.
+
+```ruby
+def self.drop_table
+  sql = "DROP TABLE IF EXISTS courses"
+  DB[:conn].execute(sql)
+end
+```
+Run `rspec` and your will see the next error, `undefined method insert`. Again, lets take a look at our `student.rb` file and see if we have reusable code. 
+
+It looks like we have an `insert` method built for us so lets modify that and use it. Let's also create a helper method called `attribute_values` to keep everything organized. Instead of typing out all of the values in our insert method, we can call the `attribute_values` method instead.
+
+```ruby
+def attribute_values
+  [name, department_id]
+end
+
+def insert
+  sql = <<-SQL
+    INSERT INTO courses
+    (name, department_id)
+    VALUES
+    (?,?)
+  SQL
+  DB[:conn].execute(sql, attribute_values)
+
+  @id = DB[:conn].execute("SELECT last_insert_rowid() FROM courses")[0][0]
+end
+```
+Now when we run `rspec`, we see `undefined method new_from_db`. This method is going to take the information from the database and create a new instance by utilizing a method called `tap`. The `tap` method "yields `self` to the block, and then returns `self`. The primary purpose of this method is to “tap into” a method chain, in order to perform operations on intermediate results within the chain." In our case we are tapping into Course and creating a new instance while assigning it's attributes to specific rows.
+
+```ruby
+def self.new_from_db(row)
+  self.new.tap do |s|
+    s.id = row[0]
+    s.name =  row[1]
+    s.department_id = row[2]
+  end
+end
+```
+
+Now we can create a Courses table, insert a course and drop a course. Being able to find a course by it's name would also be useful, which is our next error. 
```

Comment:
Maybe show how to prove this in `binding.pry`? We really gotta show them how to test things without tests!